### DISCUSSION TITLE
The type 'BitmapSource' is defined in an assembly that is not referenced

### DISCUSSION THREAD

**Discussion by s-lazukov** [unoplatform/uno#8626] [Q&A] [answered]
Hello

I have problems with the Visual Studio project

Multi-Platform App (Uno Platform|WinUI|net6)
Microsoft Visual Studio Community 2022 (64-bit) - Version 17.1.5
Win 10 Pro, 21H1

I have two projects.
I add a third one with a single class, such as `BitmapProvider` with a single method, such as `BitmapSource LoadBitmap()`. When I use this method on a Shared project, the Windows project build breaks.
The following error is thrown:
`error CS0012: The type 'BitmapSource' is defined in an assembly that is not referenced. You must add a reference to assembly 'Uno.UI, Version=255.255.255.255, Culture=neutral, PublicKeyToken=null'.`
Nuget package changed from Uno.UI to Uno.WinUI

Do you know to resolve that issue?

Thanks,
Sergey
[WinUINet6Test.zip](https://github.com/unoplatform/uno/files/8563295/WinUINet6Test.zip)

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the repro. 

To create a cross targeted library, you'll need to create a multi-platform library. You can find an example here: https://github.com/unoplatform/Uno.Samples/tree/master/UI/ControlLibrary, or create a library using `Multi-Platform Library (Uno Platform Visual Studio template).

You will need to adjust it a bit to target WinAppSDK (as the template uses UWP and not WinAppSDK yet) as follows:

```xml
<Project Sdk="MSBuild.Sdk.Extras/3.0.23">
	<!--
	Adding project references to this project requires some manual adjustments.
	Please see https://github.com/unoplatform/uno/issues/3909 for more details.
	-->
	<PropertyGroup>
		<TargetFrameworks>net6.0-windows10.0.19041.0;netstandard2.0;net6.0-ios;net6.0-macos;net6.0-maccatalyst;net6.0-android</TargetFrameworks>
		<!-- Ensures the .xr.xml files are generated in a proper layout folder -->
		<GenerateLibraryLayout>true</GenerateLibraryLayout>
	</PropertyGroup>
	<ItemGroup Condition="'$(TargetFramework)'!='net6.0-windows10.0.19041.0'">
		<PackageReference Include="Uno.WinUI" Version="4.2.6" />
	</ItemGroup>
	<ItemGroup Condition="'$(TargetFramework)'=='net6.0-windows10.0.19041.0'">
		<PackageReference Include="Microsoft.WindowsAppSDK" Version="1.0.0" />
		<PackageReference Include="Microsoft.Windows.SDK.BuildTools" Version="10.0.22000.197" />
	</ItemGroup>
	<ItemGroup>
		<Page Include="**\*.xaml" Exclude="bin\**\*.xaml;obj\**\*.xaml" />
		<Compile Update="**\*.xaml.cs">
			<DependentUpon>%(Filename)</DependentUpon>
		</Compile>
	</ItemGroup>
	<ItemGroup>
		<UpToDateCheckInput Include="**\*.xaml" Exclude="bin\**\*.xaml;obj\**\*.xaml" />
	</ItemGroup>
</Project>
```


To explain the behavior, if you create a `net6.0` library and only reference Uno.WinUI, Uno will take precedence over the WinAppSDK assemblies, causing this issue.

**Reply by s-lazukov** [reply]
It's working! Thanks a lot
